### PR TITLE
#187419010 A seller should be able to create/add items

### PR DESCRIPTION
**Pull request by cyusasnave**
### Description
As a Seller
I should be able to add/create items for sale
This page should only be accessible to sellers, i.e authenticated
Such that the stock can be seen by any requesting client
Handle all responses and errors from the backend

### How to test it?

1. Clone the repo and checkout to `feat-create-products-187419010`
2. Run `npm run dev` and go to route `/dashboard/seller/products/new`

### Pivot tracker id

187419010

### CODE REVIEW

**Review by Angemichel12 — APPROVED** [review]

**Review by Hakizimana-Clement — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -18,30 +18,39 @@
 	"dependencies": {
 		"@hookform/resolvers": "^3.6.0",
 		"@reduxjs/toolkit": "^2.2.5",
+		"@testing-library/user-event": "^14.5.2",
 		"axios": "^1.7.2",
+		"axios-mock-adapter": "^1.22.0",
```
Comment (leandreAlly) [inline package.json]: check if this library is being or used, if not remove it  "axios-mock-adapter": "^1.22.0",

File: package.json [hunk 2]
```diff
@@ -18,30 +18,39 @@
 	"dependencies": {
 		"@hookform/resolvers": "^3.6.0",
 		"@reduxjs/toolkit": "^2.2.5",
+		"@testing-library/user-event": "^14.5.2",
```
Comment (leandreAlly) [inline package.json]: let's also this be installed as dev dependency: "@testing-library/user-event": "^14.5.2",

File: package.json [hunk 3]
```diff
@@ -18,30 +18,39 @@
...
 		"react": "^18.2.0",
+		"react-dnd": "^16.0.1",
+		"react-dnd-html5-backend": "^16.0.1",
 		"react-dom": "^18.2.0",
 		"react-hook-form": "^7.51.5",
 		"react-icons": "^5.2.1",
 		"react-redux": "^9.1.2",
 		"react-router-dom": "^6.23.1",
+		"react-spinners": "^0.13.8",
+		"redux-mock-store": "^1.5.4",
```
Comment (leandreAlly) [inline package.json]: let's this also be installed as dev dependency: "redux-mock-store": "^1.5.4",

File: package.json [hunk 4]
```diff
@@ -18,30 +18,39 @@
...
 		"sonner": "^1.5.0",
 		"tailwind-merge": "^2.3.0",
 		"typical": "^7.1.1",
```
Comment (leandreAlly) [inline package.json]: this library should be removed since it's not being used: "typical": "^7.1.1",

File: src/@types/Category.tsx [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { DynamicData } from './DynamicData';
+
+export interface CategoryState {
+	isLoading: boolean;
+	categories: DynamicData[];
+	error: string | null;
+}
```
Comment (leandreAlly) [inline src/@types/Category.tsx]: this file should be .ts since it doesn't contain any jsx code.

**Review by leandreAlly — CHANGES_REQUESTED** [review]

**Review by leandreAlly — APPROVED** [review]

**Pull request merged**